RocketChat/Rocket.Chat
# fix: verifying if both app and environment is enterprise to enable

## Proposed changes (including videos or screenshots)
This PR adds verification of the "isEnterpriseOnly" flag for the app, ensuring that only enterprise apps can be enabled in an enterprise environment. The PR includes the following changes:

In the frontend code (`AppMenu.js`):

- Checks if the license is enterprise using the isEnterprise() method.
- Renders the "Enable App" button only if the app is marked as enterprise-only and the license is enterprise.

In the backend code (`:id/status` controller):

- Uses the isEnterprise() method to determine if the license is enterprise.
- Verifies if the app is marked as enterprise-only and the license is an enterprise by requesting it to the `/v1/apps/${appId}?appVersion=${version}` marketplace API endpoint.
- If both conditions are met, proceeds with enabling the app.
- If the app is enterprise-only but the environment is not valid (not enterprise or no longer enterprise), returns an error with the message "Invalid environment for enabling an enterprise app".

These changes ensure that only enterprise apps can be enabled in a valid enterprise environment, providing better control and preventing invalid configurations.





## Issue(s)


## Steps to test or reproduce


1. On an enterprise env, install some enterprise apps (Google Calendar/drive, Instagram....)
2. Change the env from enterprise to free version
3. Try to enable the app, it shouldn't be possible since the button will not be visible

With that in mind, you can test the backend part by performing a `curl`, for example:
```
curl -X POST \
   <your-env>/apps/:<app-id>/status
  -H "Content-Type: application/json" \
  -H "x-auth-token: <auth-token>" \
  -H "x-user-id: <user-id>" \
  --data-raw '{"status":"manually_enabled"}' \
  --compressed
```
If the env is not enterprise anymore you should face an error that might look like: "'Invalid environment for enabling enterprise app'"
## Further comments

AECO-275
